stdlib/public/Bionic/Glibc.swift

```diff
@@ -0,0 +1,215 @@
+//===----------------------------------------------------------------------===//
```

Comment:
@gribozavr I've been trying to combine `stdlib/public/Bionic` and `stdlib/public/Glibc`.

[`stdlib/public/Glibc/CMakeLists.txt` uses the host system name to either use a Linux or FreeBSD modulemap](https://github.com/apple/swift/blob/0e40128d1a2bfd59730d87e9f22c9f759392ca18/stdlib/public/Glibc/CMakeLists.txt#L19-L25). I figure I can do something similar to use either a Linux or Android modulemap--but instead of using the host system name (which is always Linux when cross-compiling from Linux to Android), I produce a modulemap for each requested SDK and architecture:

```cmake
# Unabbreviated source code here: https://gist.github.com/modocache/08e7aefd6c220d7750ac
foreach(SDK ${SWIFT_SDKS})
  foreach(arch ${SWIFT_SDK_${SDK}_ARCHITECTURES})
  string(TOLOWER "${SDK}" sdk)
  set(output_dir "${SWIFTLIB_DIR}/${sdk}/${arch}/glibc")

  # ...

  # Configure the modulemap based on the target. Each platform needs to
  # reference different headers, based on what's available in their glibc.
  set(modulemap_path "${CMAKE_CURRENT_BINARY_DIR}/${sdk}/${arch}/module.map")
  if("${SDK}" STREQUAL "FREEBSD")
    configure_file(module.freebsd.map.in "${modulemap_path}" @ONLY)
  elseif("${SDK}" STREQUAL "ANDROID")
    configure_file(module.android.map.in "${modulemap_path}" @ONLY)
  else()
    configure_file(module.map.in "${modulemap_path}" @ONLY)
  endif()

  # Create the lib/swift SDK/arch subdirectory and copy the configured
  # modulemap there.
  add_custom_command_target(unused_var
    COMMAND
        "${CMAKE_COMMAND}" "-E" "make_directory" "${output_dir}"
    COMMAND
        "${CMAKE_COMMAND}" "-E" "copy_if_different"
        "${modulemap_path}"
        "${output_dir}/module.map"
    CUSTOM_TARGET_NAME "copy_${sdk}_${arch}_glibc_module"
    OUTPUT "${output_dir}/module.map" "${output_dir}"
    DEPENDS "${modulemap_path}"
    COMMENT "Copying Glibc module to ${output_dir}")

  # Install the modulemap. When compiling Glibc, make sure
  # we use the correct module map for the current SDK.
  swift_install_in_component(stdlib
    FILES "${output_dir}/module.map"
    DESTINATION "${output_dir}")
  add_swift_library(swiftGlibc IS_SDK_OVERLAY
    ${sources}
    FILE_DEPENDS "copy_${sdk}_${arch}_glibc_module" "${output_dir}"
    TARGET_SDKS "${SDK}"
    SWIFT_COMPILE_FLAGS "-Xcc" "-fmodule-map-file=${modulemap_path}"
    INSTALL_IN_COMPONENT stdlib-experimental)
  endforeach()
endforeach()
```

The problem with this approach is that `swiftc` appears to expect that a Glibc modulemap exists at the following path:

```
/path/to/built/swift-linux-x86_64/lib/swift/glibc/module.map
```

Whereas the above CMake file places the module maps here:

```
/path/to/built/swift-linux-x86_64/lib/swift/linux/x86_64/glibc/module.map
/path/to/built/swift-linux-x86_64/lib/swift/android/armv7/glibc/module.map
```

This causes the following failure to occur when running tests that `import Glibc`:

```
<unknown>:0: error: missing required module 'SwiftGlibc'
```

Placing either one of these modulemaps at the expected path `lib/swift/glibc/module.map` gets the tests to pass for that target. That is, copying the Linux modulemap ensures the Linux build succeeds (and Android fails), and copying the Android modulemap causes Android to succeed and Linux to fail.

I'm not sure what the significance of the `/path/to/built/swift-linux-x86_64/lib/swift/glibc/module.map` path is, or how to hook up the compiler to use the SDK/arch subdirectory path `lib/swift/android/armv7/glibc/module.map` instead.

Any suggestions? I've been banging my head against this for a while, so any help would be greatly appreciated! :bow: